src/core/service.c

```diff
@@ -4322,6 +4324,53 @@ static int service_get_timeout(Unit *u, usec_t *timeout) {
         return 1;
 }
 
+static int bus_name_pid_lookup_callback(sd_bus_message *reply, void *userdata, sd_bus_error *ret_error) {
+        const sd_bus_error *e;
+        Unit *u = userdata;
+        uint32_t pid;
+        Service *s;
+        int r;
+
+        assert(reply);
+        assert(u);
+
+        s = SERVICE(u);
+        s->bus_name_pid_lookup_slot = sd_bus_slot_unref(s->bus_name_pid_lookup_slot);
+
+        if (!s->bus_name ||
```

Comment:
This check was not immediately obvious to me, and initially I thought you meant `!pid_is_valid()`, would be nice to have a small comment